File: 26.py
def subsets(nums):
    result = []
    path = []

    def backtrack(index):
        if index == len(nums):
            result.append(path.copy())
            return
        
        # include current element
        path.append(nums[index])
        backtrack(index + 1)

        # exclude current element
        path.pop()
        backtrack(index + 1)

    backtrack(0)
    return result

File: 26.js
function subsets(nums) {
    const result = [];
    const path = [];

    function backtrack(index) {
        if (index === nums.length) {
            result.push([...path]);
            return;
        }

        // include
        path.push(nums[index]);
        backtrack(index + 1);

        // exclude
        path.pop();
        backtrack(index + 1);
    }

    backtrack(0);
    return result;
}
